src/main/scala/io/keen/client/scala/BatchWriterClient.scala

```diff
@@ -0,0 +1,208 @@
+package io.keen.client.scala
+
+import java.util.concurrent.{ Executors, ScheduledThreadPoolExecutor, TimeUnit }
+
+import scala.collection.mutable.ListBuffer
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+import com.typesafe.config.{ Config, ConfigFactory }
+
+object BatchWriterClient {
+  val MinSendIntervalEvents: Long = 100
+  val MaxSendIntervalEvents: Long = 10000
+  val MinSendInterval: FiniteDuration = 60.seconds
+  val MaxSendInterval: FiniteDuration = 1.hour
```

Comment:
That makes sense. I'll yield until I or anyone else encounters a practical scenario where these don't work and raises an issue, but good to have your perspective.